DFImageManager/Source/Core/Utilities/DFCompositeImageManager.m

```diff
@@ -69,6 +69,14 @@ - (BOOL)canHandleRequest:(DFImageRequest *)request {
     return DFManagerForRequest(request) != nil;
 }
 
+- (BOOL)hasCacheWithRequest:(DFImageRequest *)request {
+    id<DFImageManaging> manager = DFManagerForRequest(request);
+    if (manager) {
+        return [manager hasCacheWithRequest:request];
+    }
+    return NO;
```

Comment:
Current implementation raises an exception when there is no manager registered for the request.

```objective-c
- (DFImageTask *)imageTaskForRequest:(DFImageRequest *)request completion:(DFImageRequestCompletion)completion {
    id<DFImageManaging> manager = DFManagerForRequest(request);
    if (!manager) {
        [NSException raise:NSInvalidArgumentException format:@"There are no managers that can handle the request %@", request];
    }
    return [manager imageTaskForRequest:request completion:completion];
}
```